executor/aggregate.go

```diff
@@ -977,3 +997,278 @@ func (e *groupChecker) reset() {
+	curGroupID           int
+	previousLastGroupKey string
+	firstGroupKey        []byte
+	lastGroupKey         []byte
+	firstRowDatums       []types.Datum
+	lastRowDatums        []types.Datum
+	sameGroup            []bool
+}
+
+func newVecGroupChecker(ctx sessionctx.Context, stmtCtx *stmtctx.StatementContext, items []expression.Expression) *vecGroupChecker {
+	sameGroup := make([]bool, 1024)
+	return &vecGroupChecker{
+		ctx:          ctx,
+		StmtCtx:      stmtCtx,
+		GroupByItems: items,
+		groupRowsNum: 0,
+		curGroupID:   0,
+		sameGroup:    sameGroup,
+	}
+}
+
+// splitChunk divides the chunk into more groups which the row in the same group have the same groupKey
+// the return value flag means whether the groupKey of the first group of the newly passed chunk is equal to the groupKey of the last group left before
+// TODO: Since all the group by items are only a column reference, guaranteed by building projection below aggregation, we can directly compare data in a chunk.
+func (e *vecGroupChecker) splitChunk(chk *chunk.Chunk) (flag bool, err error) {
+	numRows := chk.NumRows()
+	e.curGroupID = 0
+	e.groupRowsIndex = e.groupRowsIndex[:0]
+	e.firstGroupKey = e.firstGroupKey[:0]
+	e.lastGroupKey = e.lastGroupKey[:0]
+	e.firstRowDatums = e.firstRowDatums[:0]
+	e.lastRowDatums = e.lastRowDatums[:0]
+	if len(e.GroupByItems) == 0 {
+		e.groupRowsIndex = append(e.groupRowsIndex, numRows)
+		return true, nil
+	}
+
+	if len(e.sameGroup) < numRows {
+		e.sameGroup = make([]bool, numRows)
+	}
+	if numRows != 0 {
+		e.sameGroup[0] = false
+	}
+	for i := 1; i < numRows; i++ {
+		e.sameGroup[i] = true
+	}
+
+	for _, item := range e.GroupByItems {
+		tp := item.GetType()
+		eType := tp.EvalType()
+		col, err := expression.GetColumn(eType, numRows)
+		if err != nil {
+			return false, err
+		}
+		defer expression.PutColumn(col)
+		err = expression.VecEval(e.ctx, item, chk, col)
+		if err != nil {
+			return false, err
+		}
+
+		err = e.checkOneColumn(eType, col, numRows)
+		if err != nil {
+			return false, err
+		}
+	}
+	e.firstGroupKey, err = codec.EncodeValue(e.StmtCtx, e.firstGroupKey, e.firstRowDatums...)
+	if err != nil {
+		return false, err
+	}
+
+	e.lastGroupKey, err = codec.EncodeValue(e.StmtCtx, e.lastGroupKey, e.lastRowDatums...)
+	if err != nil {
+		return false, err
+	}
+
+	if len(e.previousLastGroupKey) == 0 {
+		flag = false
+	} else {
+		groupKey0 := string(e.previousLastGroupKey)
+		groupKey1 := string(e.firstGroupKey)
+		if groupKey0 == groupKey1 {
+			flag = true
+		} else {
+			flag = false
+		}
+	}
+	e.previousLastGroupKey = string(e.lastGroupKey)
+
+	for i := 1; i < numRows; i++ {
+		if !e.sameGroup[i] {
+			e.groupRowsIndex = append(e.groupRowsIndex, i)
+		}
+	}
+	e.groupRowsIndex = append(e.groupRowsIndex, numRows)
+	e.groupRowsNum = len(e.groupRowsIndex)
+	return flag, nil
+}
+
+func (e *vecGroupChecker) checkOneColumn(eType types.EvalType, col *chunk.Column, numRows int) (err error) {
+	previousIsNull := col.IsNull(0)
```

Comment:
We shouldn't consider this situation. See [here](https://github.com/pingcap/tidb/blob/acfc7be7941ed9f18d77009641ea77986de2ff1d/executor/aggregate.go#L892) for more details. When we use ```fetchChild()``` function , it will return when the chunk has no rows.